src/integration-test/java/io/aiven/kafka/connect/gcs/ParquetIntegrationTest.java

```diff
@@ -0,0 +1,407 @@
+        connectorConfig.put("format.output.type", "parquet");
+        connectRunner.createConnector(connectorConfig);
+
+        final var jsonMessageSchema = "{\"type\":\"struct\",\"fields\":[{\"type\":\"string\",\"field\":\"name\"}]}";
+        final var jsonMessagePattern = "{\"schema\": %s, \"payload\": %s}";
+
+        final List<Future<RecordMetadata>> sendFutures = new ArrayList<>();
+        int cnt = 0;
+        for (int i = 0; i < 10; i++) {
+            for (int partition = 0; partition < 4; partition++) {
+                final String key = "key-" + cnt;
+                final String value =
+                        String.format(
+                                jsonMessagePattern,
+                                jsonMessageSchema, "{" + "\"name\":\"user-" + cnt + "\"}"
+                        );
+                cnt += 1;
+
+                sendFutures.add(sendMessageAsync(TEST_TOPIC_0, partition, key, value));
+            }
+        }
+        producer.flush();
+        for (final Future<RecordMetadata> sendFuture : sendFutures) {
+            sendFuture.get();
+        }
+
+        // TODO more robust way to detect that Connect finished processing
+        Thread.sleep(OFFSET_FLUSH_INTERVAL_MS * 2);
+
+        final List<String> expectedBlobs = Arrays.asList(
+                getBlobName(0, 0, compression),
+                getBlobName(1, 0, compression),
+                getBlobName(2, 0, compression),
+                getBlobName(3, 0, compression));
+        assertIterableEquals(expectedBlobs, testBucketAccessor.getBlobNames(gcsPrefix));
+
+        final Map<String, List<GenericRecord>> blobContents = new HashMap<>();
+        for (final String blobName : expectedBlobs) {
+            final var records =
+                    ParquetUtils.readRecords(
+                            tmpDir.resolve(Paths.get(blobName)),
+                            testBucketAccessor.readBytes(blobName)
+                    );
+            blobContents.put(blobName, records);
+        }
+        cnt = 0;
+        for (int i = 0; i < 10; i++) {
+            for (int partition = 0; partition < 4; partition++) {
+                final var name = "user-" + cnt;
+                final String blobName = getBlobName(partition, 0, compression);
+                final var record = blobContents.get(blobName).get(i);
+                final String expectedLine = "{\"value\": {\"name\": \"" + name + "\"}}";
+                assertEquals(expectedLine, record.toString());
+                cnt += 1;
+            }
+        }
+    }
+
+    @Test
+    final void schemaChanged(@TempDir final Path tmpDir)
+            throws ExecutionException, InterruptedException, IOException {
+        final Map<String, String> connectorConfig = basicConnectorConfig();
+        final var compression = "none";
+        connectorConfig.put("format.output.fields", "value");
+        connectorConfig.put("format.output.fields.value.encoding", "none");
```

Comment:
Lets move some general config to a function, so do not disturb test writer/reader from a main task.?